dotnet/src/SemanticKernel/AI/OpenAI/Tokenizers/GPT3Tokenizer.cs

```diff
@@ -0,0 +1,194 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using Microsoft.SemanticKernel.AI.OpenAI.Tokenizers.Settings;
+
+namespace Microsoft.SemanticKernel.AI.OpenAI.Tokenizers;
+
+/// <summary>
+/// Port of GPT3 Javascript tokenizer recommended by OpenAI.
+/// See https://platform.openai.com/tokenizer and
+/// https://help.openai.com/en/articles/4936856-what-are-tokens-and-how-to-count-them
+/// </summary>
+public static class GPT3Tokenizer
+{
+    private static readonly ConcurrentDictionary<string, string> s_bpeCache = new();
```

Comment:
Examples or integration tests for using these with functions would be useful.

Reply:
what about the tests in the PR?

Reply:
I was hoping for something that would help inform Skill developers to detect when and how to use this tokenizer. Maybe integration with `SemanticTextPartitioner`

Reply:
I would do that separately. That will require detecting which model the user is using. E.g. one should not use this tokenizer in combination with Hugging Face or Bard or other models. We've been talking about having a "AI provider package" wraps all those concerns.

Reply:
I expect more here but I will resolve this conversationper @adrianwyatt request so this PR can move forward.